velox/docs/develop/connectors.rst

```diff
@@ -0,0 +1,95 @@
+==========
+Connectors
+==========
+
+Connectors allow reading and writing data to and from external sources.
+This concept is similar to `Presto Connectors <https://prestodb.io/docs/current/develop/connectors.html>`_.
+The :ref:`TableScanNode<TableScanNode>` operator reads external data via a connector.
+The :ref:`TableWriteNode<TableWriteNode>` operator writes data externally via a connector.
+The various connector interfaces in Velox are described below.
```

Comment:
To me, it's similar only by analogy.  Presto connectors differ in the following ways, which make me feel we should alter this documentation to highlight it:

* Please note I am not an expert in this codebase and would love to be corrected of any misunderstandings

1) Presto Connectors are actually Plugins.  Plugins are isolated code, shipped as JARs, that implement the SPI interfaces in Presto.  Plugins may be versioned separately from the rest of Presto--as long as SPI is backwards compatible, people don't need to upgrade their plugins.  Velox connectors do not appear to have a plugin concept.  It appears they are built by a code level dependency on Velox--`registerConnector` actually seems to add a new connector to a map in the Velox binary.  This suggests to me that connectors are static.  Therefore, I think it may be confusing to compare these two similarly without also noting the differences.
2) There's obviously a much smaller interface footprint in Velox, since it's only concerned with datasinks, dataproviders and split deserialization.

Also, I think it would be helpful (going back to #1) to note precisely how one registers a connector--I couldn't find it here.  Overall, writing for the perspective of someone who wants to add a new connector may be helpful.

Reply:
@tdcmeehan thanks for the feedback. You are right about the plugin difference. Velox connectors are not plugins.
We should have a toy connector example to demonstrate the registration. I will follow up with another PR to highlight these differences.